### PR TITLE
Fix issues with numbers, adjectives, and named entities

### LINKED ISSUE #29: Handle cardinals in nummod dependency
a nummod is a number modifier.

When a word has a nummod as a modifier of another modifier, we should treat the word and the nummod as one unit. For example, for "He is a five star general," the definition "star general" doesn't really make sense. 

Nummod should be ignored entirely if it directly modifies definition, for example, in "He is one of the three piglets."

### LINKED ISSUE #37: Handle ordinals
"...was the first Australian Flying Corps unit to see active service."

### LINKED ISSUE #60: Handle cardinals in typeword position
"Mu Dan was one of the most important poets of 20th century China"

### PR DESCRIPTION
**Pull request by moldot**
Remove nummod that directly modifies definition, and merge nummod
with its parent of parent is not a definition head.

UPDATE: 01/25/16:
- Ignore numeric modifiers and ordinal in the output, but still keep it in the internal representation. 
```
sentence = u'The 2001 French Open was the second Grand Slam event of 2001 and the 100th edition of the French Open.'
expected = [u'event', u'grand slam event', u'2001 event', u'edition', u'french open edition']
```

- When a number or an adjective has an of-modifier, raise that of modifier and make the number/adjective a modifier instead. 
```
'...the largest of the three pyramids'
[def: largest [of: pyramids [nummod: three]]] --> [def: pyramids [nummod: three] [mod: largest]]
[def: two [of: companies [mod: successful]]] --> [def: company [mod: successful] [nummod: two]]
```

- We also found out, as we are dealing with numbers, that named entities have been handled incorrectly. Previously, when we encounter a node that is the root of a named entity, we fetch the entire entity and stop traversing the dependency tree further. This is to make sure that the entire entity is contained in one node rather than split into multiple nodes. For example, we don't want:
```
'of the united states' --> [of: states [mod: united]]
```

However, such a strategy causes problem when when a named entity has modifiers that are not part of the entity, such as 'United States House of Representatives', where 'United States' and 'House of Representatives' are two separate named entities. 

So, we now make it so that named entity can be recognize from any node, not necessarily the root. This is necessary because dependency tree is sometimes not consistent with NE recognition, and therefore the NE root node might not actually be the parent of the entity. We also look for modifiers of named entity as well, without splitting a named entity in to multiple nodes.
```
sentence = u"Levi Warner was a Democratic member of the United States House of Representatives from Connecticut's 4th congressional district from 1876 to 1879."
 expected = [u'member', u'house of representatives member', u'united states house of representatives member', u'democratic member', u'democratic']
```
In the previous example, 'the' is the root of 'the united states'.

Closes #29 , closes #37  , closes #60

### CODE REVIEW

**Comment by moldot** [comment]
Don't review yet. Will look more into this.

File: tests/test_assertion_definitions.py [hunk 1]
```diff
@@ -318,10 +333,81 @@ def test_verb_like_subject_simplified(self):
...
+    def test_cardinal_number(self):
+        sentence = u'Waskaganish is a Cree community of over 2,200 people at the mouth of the Rupert River on the south-east shore of James Bay in the Eeyou Istchee territory in Northern Quebec, Canada.'
+        expected = [u'community', u'cree community', u'people community']
+        actual, entry = get_assertion_definitions(sentence, get_rep=True)
+        self.assertItemsEqual(expected, actual)
+        self.assertTrue(u'[nummod: 2,200' in entry)
+
+    def test_one_of(self):
+        sentence = u'The Pleiad Electra of Greek mythology was one of the seven daughters of Atlas and Pleione.'
+        expected = [u'daughter', u'atlas daughter', u'pleione daughter']
```
Comment (alvaromorales) [inline tests/test_assertion_definitions.py]: what happens to capitalization? I disabled lemmatizing on the web demo, and the output definitions are all lowercased. Let's not lose true-casing (I see that the internal representation stores tokens in true case)

File: whoami/defextract.py [hunk 2]
```diff
@@ -27,33 +27,39 @@ def pobj_definitions(head):
 def build_head_modifier_tree(head, relation):
     global ners
 
+    print 'building'
```
Comment (alvaromorales) [inline whoami/defextract.py]: remove?

File: tests/test_assertion_definitions.py (hunk 1 above)
Comment (moldot) [inline tests/test_assertion_definitions.py]: This somewhat follows original design that the output will be in lowercase, which can be easily changed. The lemma flag dictates lemmatization, for example changing 'princesses' to 'princess'. It doesn't prohibits 'Princesses' to become 'princesses'. 

What do you feel about adding another flag 'lowercase' to force lowercase output? Anyhow, let's open this as a separate issue.

Comment (alvaromorales) [inline tests/test_assertion_definitions.py]: +1 to adding the flag and opening a separate issue!

**Comment by alvaromorales** [comment]
lgtm :ship:

**Pull request merged**